core/node/node_sync/src/batch_transaction_updater/mod.rs

```diff
@@ -0,0 +1,302 @@
+    ) -> Self {
+        Self {
+            sl_client,
+            l1_transaction_verifier: L1TransactionVerifier::new(diamond_proxy_addr, pool.clone()),
+            pool,
+            health_updater: ReactiveHealthCheck::new("batch_transaction_updater").1,
+            sleep_interval,
+            processing_batch_size,
+        }
+    }
+
+    pub fn health_check(&self) -> ReactiveHealthCheck {
+        self.health_updater.subscribe()
+    }
+
+    async fn apply_status_update(
+        &self,
+        connection: &mut Connection<'_, Core>,
+        db_eth_history_id: u32,
+        receipt: TransactionReceipt,
+        l1_block_numbers: &L1BlockNumbers,
+    ) -> anyhow::Result<()> {
+        let updated_status =
+            l1_block_numbers.get_finality_status_for_block(receipt.block_number.unwrap().as_u32());
+
+        if updated_status == EthTxFinalityStatus::Pending {
+            // Errors from apply_status_update will result in a warning and retry
+            anyhow::bail!(
+                "Transaction {} is still pending on SL",
+                receipt.transaction_hash
+            );
+        }
+
+        let db_eth_history_tx = connection
+            .eth_sender_dal()
+            .get_eth_tx_history_by_id(db_eth_history_id)
+            .await?;
+
+        let batch_number = connection
+            .blocks_dal()
+            .get_l1_batch_number_by_eth_tx_id(db_eth_history_tx.eth_tx_id)
+            .await?
+            .expect("eth_tx_history row must match a l1 batch");
+
+        // validate the transaction against db
+        match db_eth_history_tx.tx_type {
+            AggregatedActionType::Commit => {
+                self.l1_transaction_verifier
+                    .validate_commit_tx(&receipt, batch_number)
+                    .await?
+            }
+            AggregatedActionType::PublishProofOnchain => {
+                self.l1_transaction_verifier
+                    .validate_prove_tx(&receipt, batch_number)
+                    .await?
+            }
+            AggregatedActionType::Execute => {
+                self.l1_transaction_verifier
+                    .validate_execute_tx(&receipt, batch_number)
+                    .await?
+            }
+        };
+
+        connection
+            .eth_sender_dal()
+            .confirm_tx(
+                receipt.transaction_hash,
+                updated_status,
+                receipt.gas_used.unwrap_or_default(), // we don't care about gas used for synced transaction
+            )
+            .await?;
+
+        tracing::info!(
+            "Updated finality status for transaction {} ({} for batch {}) from {:?} to {:?}",
+            db_eth_history_tx.tx_hash,
+            db_eth_history_tx.tx_type,
+            batch_number,
+            db_eth_history_tx.eth_tx_finality_status,
+            updated_status
+        );
+
+        Ok(())
+    }
+
+    async fn update_statuses(
+        &self,
+        mut connection: Connection<'_, Core>,
```

Comment:
You're holding this connection across multiple calls to RPC clients, which is a general anti-pattern. Is it supposed to be transactional (AFAICT, it's not now)? Otherwise, it makes sense to acquire short-living connections as they are needed inside the method.